### PR TITLE
Top Menu

### PR DESCRIPTION
**Pull request by retif**
This PR implements a new top menu design.

## Steps to test:

- [ ] open virtual y page
- [ ] see new virtual y menu under primary menu
- [ ] check that at virtual y main page and video page top menu works in dark mode
- [ ] at the admin panel at virtual y settings page you can change colors for links and background for both light and dark mode

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [ ] This change does not contain front-end fixes.
- [ ] I have flagged this PR "Needs Review" or pinged the VY devs/QA team in Slack

### CODE REVIEW

**Comment by retif** [comment]
retest

File: modules/openy_gc_demo/README.md [hunk 1]
```diff
@@ -3,4 +3,5 @@ For creating demo content execute these commands:
 ```shell script
 drush mim --group=openy_demo_term
 drush mim --group=virtual_y
+drush mim migrate_plus.migration.virtual_y_menu_link_main
```
Comment (anpolimus) [inline modules/openy_gc_demo/README.md:6]:
```suggestion
drush mim virtual_y_menu_link_main
```

**Review by anpolimus — CHANGES_REQUESTED** [review]

**Comment by anpolimus** [comment]
image
Mobile menu of Virtual Y should be aligned with mobile menu of the Open Y

**Comment by anpolimus** [comment]
The same for tablet
image

**Comment by anpolimus** [comment]
At design, mobile menu and website menu has a similar size

File: js/gated-content/src/scss/components/_top-menu.scss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    button {
+      @include median(display, none, inline-block);
+      outline:          none;
+      border:           none;
+      background-color: transparent;
+      text-align:       right;
+
+      &:before {
+        display:     inline-block;
+        font-family: 'Font Awesome 5 Free';
```
Comment (NightHunterSV) [inline js/gated-content/src/scss/components/_top-menu.scss:41]: Not sure this will work in all the OpenY themes

Comment (retif) [reply js/gated-content/src/scss/components/_top-menu.scss]: replaced this icon with svg

File: js/gated-content/src/scss/components/_top-menu.scss [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      @include median(display, block);
+      border:        1px solid $gray;
+      border-radius: 5px;
+      padding:       15px;
+      text-align:    center;
+      color:         $gray;
+    }
+
+    > div {
+      display:               grid;
```
Comment (retif) [inline js/gated-content/src/scss/components/_top-menu.scss:73]: this grid is mobile only, so no ie compatibility required

**Review by hamrant — CHANGES_REQUESTED** [review]
We need hook_update for `openy_gated_content.settings.yml`, please update only `top_menu` param.

**Review by hamrant — APPROVED** [review]

**Review by NightHunterSV — APPROVED** [review]

File: assets/js/openy_rose.behaviors.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+(function ($) {
+  "use strict";
+
+  /**
+   * Set css class 'primary-menu-minimize' for body element after primary menu affix activated on scroll
+   */
+  Drupal.behaviors.OpenyRoseAffixClassForPrimaryMenu = {
+    attach: function (context, settings) {
+      if ($('body').hasClass('openy_rose-based')) {
```
Comment (anpolimus) [inline assets/js/openy_rose.behaviors.js:12]: Please, try to use hoook_preprocess_html instead of this

**Review by anpolimus — CHANGES_REQUESTED** [review]

**Comment by NadyaRudenko** [comment]
- please add hover state on Lily

- please change hover state to be as here https://xd.adobe.com/view/e9d9a7ca-1747-4b26-6f69-79dcd76ecadf-9fe8/screen/2178740a-1dbb-49d1-84ac-e64c3c5ecf9e/ for light mode. for dart mode, I guess it should be white underline, but will be clarified in standup
now underline for hover state looks this way on Rose and Carnation:
image
- close icon is bigger than in design﻿
image

- when the main menu is opened on Carnation I can see the secondary menu
image

- and on Lily I can see a thin black line﻿
image

**Comment by NadyaRudenko** [comment]
Retested, mobile menu fixed, 'cross' fixed, new underline added.

But first two items in the menu are underlined as default without hovering over them (all themes)
image
And the main menu on Carnation got effected and when it is opened I can see a hamburger menu instead of cross
image
@retif, can you please check? thanks!

**Comment by NadyaRudenko** [comment]
underlined items in mobile menu look thinner
image

**Comment by NadyaRudenko** [comment]
Retested, fixed. Thank you! 🙂

**Pull request merged**